packages/zowe-explorer/src/uss/actions.ts

```diff
@@ -47,18 +50,32 @@ export async function createUSSNode(
     nodeType: string,
     isTopLevel?: boolean
 ) {
+    await ussFileProvider.checkCurrentProfile(node);
```

Comment:
The changes in this file deal with creating a new USS file or folder, which seems unrelated to the rest of this PR. It does seem to work well, so I guess we can leave it in this time, but going forward, I would recommend making a separate PR for separate fixes/features.

This adds an extra step in the folder/file creation process, but it does give the user a more obvious way to control where they are creating the file/folder, and helps avoid errors if the user wants to do the creation without having entered a search prompt first. The fact that it's pre-filled makes helps it more painless. I like that it gets rid of the USS tree collapsing upon node creation.

Reply:
This was added to fix a bug with using the commands via the context menu, instead of via the button on the toolbar...it should also allow us to create new through the command palette later on, but that isn't in this PR.